aten/src/ATen/native/quantized/cpu/kernels/QuantizedOpKernels.cpp

```diff
@@ -192,6 +193,42 @@ void qrelu6_kernel(const Tensor& qx, Tensor& qy) {
   });
 }
 
+void qtanh_kernel(const Tensor& qx, Tensor& qy) {
+  int64_t zero_point = qx.q_zero_point();
+  float scale = qx.q_scale();
+  float inv_scale = 1.0f / scale;
+  auto scale_vec = Vec256<float>(scale);
+  auto zero_point_vec = Vec256<float>((float)zero_point);
+  auto scale_neg_zp_premul_vec = scale_vec * zero_point_vec.neg();
+
+  AT_DISPATCH_QINT_TYPES(qx.scalar_type(), "qtanh", [&]() {
+    qy = at::_empty_affine_quantized(
+        qx.sizes(),
+        at::device(kCPU).dtype(SCALAR_TYPE),
+        qx.q_scale(),
+        qx.q_zero_point(),
+        qx.suggest_memory_format());
+    auto iter = TensorIterator::unary_op(qy, qx);
+
+    using Vec = Vec256<scalar_t>;
+    cpu_kernel_vec(
+      iter,
+      [&](scalar_t value_qx) -> scalar_t {
+        const auto value_dx = at::dequantize_val(scale, zero_point, value_qx);
+        return at::quantize_val<scalar_t>(scale, zero_point, std::tanh(value_dx));
+      },
+      [&](Vec value_qx) -> Vec {
+        const auto value_dx = value_qx.dequantize(scale_vec, zero_point_vec, scale_neg_zp_premul_vec);
+        Vec::float_vec_return_type retvals;
+        for (int idx = 0; idx < Vec::float_num_vecs(); ++idx) {
```

Comment:
Lets go for a quantized tanh implementation and remove the need to call float tanh here for speed. (@hx89: Do we have a look up table based implementation ready?).
We should eventually have a lookup table that takes in an input in the range [-4,4] and returns values in the range  [-1 ,1]. tanh(4) = 0.9993. The table itself is 8 bits wide. We will need to determine the size of the table via experiments. This can occur in a later PR if we dont have a quantized implementation ready.

Reply:
I don't think the float here has a huge implication, as in the end of the day it uses the native `tanh` implementation. But, you are probably right -- the LUT will be faster. However, I don't think it will be scalable -- how are we going to handle the `qint32` or `qint16` down the road?

Reply:
Yes @raghuramank100 , we have LUT based implementation for tanh in DNNLOWP: https://github.com/pytorch/pytorch/blob/master/caffe2/quantization/server/tanh_dnnlowp_op.cc

Reply:
Yes, we have LUT implementation of tanh in DNNLOWP:
https://github.com/pytorch/pytorch/blob/master/caffe2/quantization/server/tanh_dnnlowp_op.cc
If higher accuracy is needed we can use Taylor series to add more terms. We have an internal LUT implementation for float16 tanh using Taylor series.

Reply:
Yes, we have LUT implementation of tanh in DNNLOWP:
https://github.com/pytorch/pytorch/blob/master/caffe2/quantization/server/tanh_dnnlowp_op.cc
If higher accuracy is needed we can use Taylor series to add more terms. We have an internal LUT implementation for float16 tanh using Taylor series.

Reply:
How does LUT (8 and 16 bit) compare to `std::tanh`?

Reply:
I think we can optimize later, I don't feel this would be a bottleneck of the network